### PR TITLE
feat(dli): add a new data source to query enhanced connection list

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

1. Add zero value judgment to FormatTimeStampRFC3339 method.
2. Add a new data source to get the list of DLI enhanced datasource connection.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new datasource to get the list of DLI enhanced datasource connection.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDatasourceConnections_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDatasourceConnections_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceConnections_basic
=== PAUSE TestAccDatasourceConnections_basic
=== CONT  TestAccDatasourceConnections_basic
--- PASS: TestAccDatasourceConnections_basic (313.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       313.333s
```

### CODE REVIEW

File: docs/data-sources/dli_datasource_connections.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+* `name` - The name of the datasource connection.
+
+* `queues` - List of queue associated with the datasource connection.
+  The [queues](#datasource_connections_queues) structure is documented below.
+
+* `elastic_resource_pools` - List of resource pool associated with the datasource connection.
+  The [elastic_resource_pools](#datasource_connections_resource_pools) structure is documented below.
+
+* `routes` - List of route.
```
Comment (Lance52259) [inline docs/data-sources/dli_datasource_connections.md]:
```suggestion
* `routes` - List of routes.
```
please check the other places.

File: huaweicloud/services/dli/data_source_huaweicloud_dli_datasource_connections.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+	for i, v := range curArray {
+		rst[i] = map[string]interface{}{
+			"id":                     utils.PathSearch("id", v, nil),
+			"name":                   utils.PathSearch("name", v, nil),
+			"status":                 utils.PathSearch("status", v, nil),
+			"queues":                 flattenQueues(utils.PathSearch("available_queue_info", v, make([]interface{}, 0))),
+			"vpc_id":                 utils.PathSearch("dest_vpc_id", v, nil),
+			"subnet_id":              utils.PathSearch("dest_network_id", v, nil),
+			"is_privis":              utils.PathSearch("isPrivis", v, nil).(bool),
+			"created_at":             utils.FormatTimeStampRFC3339(int64(utils.PathSearch("create_time", v, nil).(float64))/1000, false),
```
Comment (Lance52259) [inline huaweicloud/services/dli/data_source_huaweicloud_dli_datasource_connections.go]: assert for nil value will cause a panic.

File: docs/data-sources/dli_datasource_connections.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/data-sources/dli_datasource_connections.md]: fixed

File: huaweicloud/services/dli/data_source_huaweicloud_dli_datasource_connections.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/data_source_huaweicloud_dli_datasource_connections.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDatasourceConnections_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDatasourceConnections_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceConnections_basic
=== PAUSE TestAccDatasourceConnections_basic
=== CONT  TestAccDatasourceConnections_basic
--- PASS: TestAccDatasourceConnections_basic (317.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       317.303s

File: docs/data-sources/dli_datasource_connections.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+<a name="datasource_connections_queues"></a>
+The `queues` block supports:
+
+* `name` - The queue name.
+
+* `id` - The peer ID of the datasource connection.
+
+* `status` - The status of the peering connection to which the datasource connection belongs.
+
+* `error_msg` - Error message when the Peering connection status is falied.
```
Comment (Lance52259) [inline docs/data-sources/dli_datasource_connections.md]:
```suggestion
* `status` - The status of the peering connection to which the datasource connection belongs.

* `error_msg` - Error message when the peering connection status is falied.
```

File: docs/data-sources/dli_datasource_connections.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+* `name` - The name of the elastic resource pool.
+
+* `id` - The peer ID of the datasource connection.
+
+* `status` - The status of the peering connection to which the datasource connection belongs.
+
+* `error_msg` - Error message when the Peering connection status status is falied.
+
+* `updated_at` - The latest update time of the elastic resource pool .
```
Comment (Lance52259) [inline docs/data-sources/dli_datasource_connections.md]:
```suggestion
* `updated_at` - The latest update time of the elastic resource pool.
```

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDatasourceConnections_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDatasourceConnections_basic -timeout 360m -parallel 4
=== RUN TestAccDatasourceConnections_basic
=== PAUSE TestAccDatasourceConnections_basic
=== CONT TestAccDatasourceConnections_basic
--- PASS: TestAccDatasourceConnections_basic (317.25s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli 317.303s

File: docs/data-sources/dli_datasource_connections.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+<a name="datasource_connections"></a>
+The `connections` block supports:
+
+* `created_at` - The creation time of the datasource connection.
+
+* `hosts` - List of the user-defined hosts information.
+  The [hosts](#datasource_connections_hosts) structure is documented below.
+
+* `id` - The ID of the datasource connection.
```
Comment (ShiChangkuo) [inline docs/data-sources/dli_datasource_connections.md]: it's better to make `id` as the first attribute in a block.

Comment (wuzhuanhong) [reply docs/data-sources/dli_datasource_connections.md]: fixed

**Pull request merged**